File: app/src/main/java/com/uber/okbuck/example/DummyComponent.java
package com.uber.okbuck.example;

import com.uber.okbuck.example.dummylibrary.AndroidModule;
import com.uber.okbuck.example.javalib.JavaModule;
import dagger.Component;

@Component(modules = {JavaModule.class, AndroidModule.class})
public interface DummyComponent {
  void inject(MainActivity activity);
}

File: libraries/kotlinandroidlibrary/src/main/kotlin/com/uber/okbuck/kotlin/di/DummyComponent.kt
package com.uber.oknuck.kotlin.di

import com.uber.oknuck.kotlin.KotlinAndroidApplication
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [AndroidSupportInjectionModule::class])
interface DummyComponent : AndroidInjector<KotlinAndroidApplication> {
    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<KotlinAndroidApplication>()
}
